src/components/InputField.js

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+
+export default class InputField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.value
+    };
+  }
+
+  updateInputField(e) {
+    this.setState({value: e.nativeEvent.target.value});
+  }
+
+  render() {
+    return (
+        <input type="text" onBlur={() => this.props.callBack()} value={this.state.value} onChange={(e) => this.updateInputField(e)}/>
```

Comment:
So just to keep us from constantly pinging the backend, what happens is that onChange (aka anytime the text value changes in the input field) we'll update this components state, but won't send it to the backend. onBlur (aka when the user leaves the input field, or otherwise clicks away) then we'll set the overall app state and send the value back to the backend. So in theory it should give you the experience you just described.